src/openforms/submissions/tests/test_on_completion_retry_chain.py

```diff
@@ -235,6 +232,11 @@ def test_backend_registration_succeeds(self, mock_update_payment):
             form__registration_backend_options={"zgw_api_group": zgw_group.pk},
             # registration failed, so an internal reference was created
             public_registration_reference="OF-1234",
+            form__product__price=10,
+            form__payment_backend="demo",
+        )
+        SubmissionPaymentFactory.create(
+            submission=submission, amount=10, status=PaymentStatus.completed
```

Comment:
This needed to be added because on line 263 we check if the update payment task is called (but now this is skipped if no payment is required)